Changelog: Add 1.0 changes

jku opened the pull request:
Leaving as draft for now:
* I believe I've listed all relevant changes so far (some trivial PRs are not listed) but obviously there are likely to be a few more changes before 1.0 
* The text section could be improved, but I'll do that with a fresher brain

Commit:
Changelog: Add 1.0 changes

Signed-off-by: Jussi Kukkonen <jkukkonen@vmware.com>

ivanayov commented:
Should we add bug fixes?
* [Metadata API: Role keyids is not ordered #1478](https://github.com/theupdateframework/python-tuf/issues/1478) /[role keyids still not ordered #1752](https://github.com/theupdateframework/python-tuf/issues/1752)
* [ngclient: should clean temp file if write fails #1575](https://github.com/theupdateframework/python-tuf/issues/1575)
* Not sure if this was present before 0.20.0: [ngclient: throws securesystemslib errors #1761](https://github.com/theupdateframework/python-tuf/issues/1761)

ivanayov commented on docs/CHANGELOG.md:
```diff
@@ -1,5 +1,43 @@
+  #1767, #1803, #1830, #1832, #1837, #1839)
+* build: Stop supporting EOL Python 3.6 (#1783)
+* build: Update dependencies (#1827, #1834)
+
```
Waiting for latest updates on #1852 as `securesystemslib` updates are already present.

jku replied:
this was added as 1861, thanks

ivanayov commented on docs/CHANGELOG.md:
```diff
@@ -1,5 +1,43 @@
+
+### Added
+* tests: Extend client testing (#1689, #1703, #1711, #1728, #1735, #1738,
+  #1742, #1766, #1777, 1831)
```
Missing hash sign: #1831

ivanayov reviewed (commented):
Seems like almost everything is present. Leaving some minor comments.

Should we add bug fixes?
* [Metadata API: Role keyids is not ordered #1478](https://github.com/theupdateframework/python-tuf/issues/1478) /[role keyids still not ordered #1752](https://github.com/theupdateframework/python-tuf/issues/1752)
* [ngclient: should clean temp file if write fails #1575](https://github.com/theupdateframework/python-tuf/issues/1575)
* Not sure if this was present before 0.20.0: [ngclient: throws securesystemslib errors #1761](https://github.com/theupdateframework/python-tuf/issues/1761)

jku commented:
> Should we add bug fixes?

They should be in the list, just with the PR number and not the issue number (I can see an argument for using issue numbers: it's just potentially a lot more work since `git log --merges v0.20.0..origin/develop` gives a good initial list of PRs, but a complete list of issues is much harder to collect). 

Out of your list all others have PRs in the list except 1754 and 1478: the latter was "fixed" last summer already but the former does seem to have the wrong issue number in the changelog. Thanks!

I fixed that and rebased to include the latest PRs 1850, 1848, 1855, 1809.

ivanayov commented:
It looks good this way as well (i.e. without specific "Bug fixes" section).

I think that after having all latest required PRs merged (including the not yet opened for #1852) this should be good to go.

Commit:
docs: Add latest changes to 1.0 changelog

Signed-off-by: Jussi Kukkonen <jkukkonen@vmware.com>

jku commented:
rebased, added a commit that documents the new PRs in that rebase (this can be squashed, just trying to make this easier to review)

ivanayov commented:
Looks good to me.

Commit:
docs: Add latest changes in 1.0 changelog

Signed-off-by: Jussi Kukkonen <jkukkonen@vmware.com>

jku commented:
rebased again, added a commit that documents the new merges in that rebase (this can be squashed, just trying to make this easier to review)

MVrachev reviewed (approved):
I had a look at all commits after release `0.20.0` and it seems that all changes are here.
LGTM!

lukpueh commented:
Superseded by #1873
